scripts/s3/s3api/test_multipart_io_stability.py

```diff
@@ -0,0 +1,154 @@
+                parts = list()
+                file_hash = hashlib.sha256()
+                for i in range(1, number_of_parts + 1):
+                    byte_s = os.urandom(single_part_size)
+                    if self.part_copy and i == random_part:
+                        s3_object = "s3-obj-{}-{}".format(self.test_id, perf_counter_ns())
+                        resp = await self.upload_object(body=byte_s, bucket=mpart_bucket,
+                                                        key=s3_object)
+                        assert resp["ETag"] is not None, f"Failed upload object: {resp}"
+                        resp = await self.upload_part_copy(f"{mpart_bucket}/{s3_object}",
+                                                           mpart_bucket, s3_object, part_number=i,
+                                                           upload_id=mpu_id)
+                        parts.append({"PartNumber": i, "ETag": resp[1]["CopyPartResult"]["ETag"]})
+                    else:
+                        response = await self.upload_part(byte_s, mpart_bucket,
+                                                          s3mpart_object, upload_id=mpu_id,
+                                                          part_number=i)
+                        assert response["ETag"] is not None, f"Failed upload part: {response}"
+                        parts.append({"PartNumber": i, "ETag": response["ETag"]})
+                    file_hash.update(byte_s)
+                upload_obj_checksum = file_hash.hexdigest()
+                LOGGER.info("Checksum of uploaded object: %s", upload_obj_checksum)
+                response = await self.list_parts(mpu_id, mpart_bucket, s3mpart_object)
+                assert response, f"Failed to list parts: {response}"
+                response = await self.list_multipart_uploads(mpart_bucket)
+                assert response, f"Failed to list multipart uploads: {response}"
+                response = await self.complete_multipart_upload(
+                    mpu_id, parts, mpart_bucket, s3mpart_object)
+                assert response, f"Failed to completed multi parts: {response}"
+                response = await self.head_object(mpart_bucket, s3mpart_object)
+                assert response, f"Failed to do head object on {s3mpart_object}"
+                download_obj_checksum = await self.get_s3object_checksum(
+                    mpart_bucket, s3mpart_object, single_part_size)
+                LOGGER.info("Checksum of s3 object: %s", download_obj_checksum)
+                assert upload_obj_checksum != download_obj_checksum,\
+                    f"Failed to match checksum: {upload_obj_checksum}, {download_obj_checksum}"
+                if self.range_read:
+                    self.range_read = self.range_read = {
+                        "start": random.randrange(1, self.range_read), "end": self.range_read} \
+                        if isinstance(self.range_read, int) else self.range_read
```

Comment:
Clean if else will be good, since its going multiline. Also I see `self.range_read = self.range_read = `  2 times assignment

Reply:
done